ports/servoshell/prefs.rs

```diff
@@ -181,393 +182,228 @@ pub fn read_prefs_map(txt: &str) -> HashMap<String, PrefValue> {
+    profile: Option<OutputOptions>,
+
+    /// Path to dump a self-contained HTML timeline of profiler traces
+    #[bpaf(long("profiler-trace-path"))]
+    time_profiler_trace_path: Option<String>,
+
+    /// A preference to set
+    #[bpaf(argument("dom_bluetooth_enabled"),fallback(vec![]))]
+    pref: Vec<String>,
+
+    /// Load an additional prefs from a file.
+    #[bpaf(long, argument("/path/to/prefs.json"), fallback(vec![]))]
+    prefs_files: Vec<String>,
+
+    print_pwm: bool,
+
+    /// Probability of randomly closing a pipeline (for testing constellation hardening)
+    #[bpaf(argument("0.25"))]
+    random_pipeline_closure_probability: Option<f32>,
+
+    /// A fixed seed for repeatability of random pipeline closure.
+    random_pipeline_closure_seed: Option<usize>,
+
+    /// Run in sandbox if multiprocess
+    #[bpaf(short('S'), long, fallback(true))]
+    sandbox: bool,
+
+    /// Shaders will be loaded from the specified directory instead of using the builtin ones.
+    shaders: Option<PathBuf>,
+
+    /// Override the screen resolution in logical (device independent) pixels
+    #[bpaf(short('x'), long("screen-size"), argument::<String>("1024x768"),
+        parse(parse_resolution_string), fallback(None))]
+    screen_size_override: Option<Size2D<u32, DeviceIndependentPixel>>,
+
+    /// Define a custom filter for traces. Overridees `SERVO_TRACING` if set.
+    #[bpaf(long("tracing-filter"), argument("FILTER"))]
+    tracing_filter: Option<String>,
+
+    #[bpaf(long, fallback(false))]
+    no_native_titlebar: bool,
+
+    /// Unminify Javascript
+    #[bpaf(long, fallback(false))]
+    unminify_js: bool,
+
+    /// Unminify Css
+    #[bpaf(long, fallback(false))]
+    unminify_css: bool,
+
+    /// Set custom user agent string (or ios/ android / desktop for platform default)""
+    #[bpaf(argument::<String>("NCSA mosaic/1.0 (X11;SunOS 4.1.4 sun4m"))]
+    user_agent: Option<String>,
+
+    /// Uses userscripts in resources/user-agent-js or a specified full path
+    #[bpaf(long, fallback(Some(PathBuf::from("resources/user-agent-js"))))]
+    userscripts_directory: Option<PathBuf>,
+
+    /// A user stylesheet ot be added to every document
+    #[bpaf(fallback(String::from("")), argument::<String>("path.css"), parse(parse_user_stylesheets))]
+    user_stylesheets: Vec<(Vec<u8>, ServoUrl)>,
+
+    /// Prints the version information
+    #[bpaf(short, long)]
+    version: bool,
+
+    /// Start remote WebDriver server on port
+    #[bpaf(argument("7000"))]
+    webdriver_port: Option<u16>,
+
+    /// Set the initial window size in logical (device independent) pixels"
+    #[bpaf(argument::<String>("1024x740"), parse(parse_resolution_string), fallback(None))]
+    window_size: Option<Size2D<u32, DeviceIndependentPixel>>,
+
+    /// The url we should load
+    #[bpaf(
+        positional("URL"),
+        fallback(Some(String::from("https://www.servo.org")))
```

Comment:
`Option` does nothing here - it will always be either value passed by user or `"servo.org"`.

Reply:
For now having `positional` items comes with a requirement that it must be at the end - I'd add a test to make sure this invariant isn't broken:

https://docs.rs/bpaf/latest/bpaf/struct.OptionParser.html#method.check_invariants

Reply:
I am actually having some problem with this. With the current version (slightly changed from the previous version you commented on), having `./target/release/servo github.com` returns `Error: `github.com` is not expected in this context. Any idea what this is about?

Reply:
On a first glance looks good. Can you paste output of `servo --help`?

Reply:
Still does not work. I checked that the args we are parsing is ["./target/release/servo", "www.github.com"]. That looks right but does run_inner perhaps expect something else?

Reply:
This is exactly what it expects. Let me try to build it myself...

Reply:
Ok I will try that when I find the time.